sdk/resource/resource.go

```diff
@@ -51,29 +74,71 @@ func Merge(a, b *Resource) *Resource {
 	if b == nil {
 		return a
 	}
-	res := &Resource{
-		labels: map[core.Key]core.Value{},
+
+	n := len(a.sorted)
+	if len(b.sorted) > len(a.sorted) {
+		n = len(b.sorted)
+	}
+	// At a minimum the merge will be as large as the largest resource.
+	s := make([]core.KeyValue, 0, n)
+	k := make(map[core.Key]struct{}, n)
+	ai, bi := 0, 0
+	for ; ai < len(a.sorted) && bi < len(b.sorted); ai, bi = ai+1, bi+1 {
```

Comment:
It's not clear to me why this is split into two parts.  Or, rather, I get why lines 100-113 exist given that this loop will not copy all attributes if one resource has more than the other.  But why this loop in the first place?  It seems like the separate iteration over `a.sorted` and then `b.sorted` would accomplish the same thing.

Reply:
Yeah, I think I'd second that - I'd drop this loop and iterate separately over `a.sorted` and then over `b.sorted`.